.github/workflows/main.yml

```diff
@@ -110,8 +110,33 @@ jobs:
           ./artifacts/package
         if-no-files-found: error
 
+    - name: Install PowerShell (Linux/macOS)
+      if: ${{ runner.os != 'Windows' }}
+      shell: bash
+      run: |
+        # Install PowerShell on Linux
+        if [[ "${{ runner.os }}" == "Linux" ]]; then
+          # Update the list of packages
+          sudo apt-get update
+          # Install pre-requisite packages
+          sudo apt-get install -y wget apt-transport-https software-properties-common
+          # Download the Microsoft repository GPG keys
+          wget -q "https://packages.microsoft.com/config/ubuntu/$(lsb_release -rs)/packages-microsoft-prod.deb"
+          # Register the Microsoft repository GPG keys
+          sudo dpkg -i packages-microsoft-prod.deb
+          # Delete the the Microsoft repository GPG keys file
+          rm packages-microsoft-prod.deb
+          # Update the list of packages after we added packages.microsoft.com
+          sudo apt-get update
+          # Install PowerShell
+          sudo apt-get install -y powershell
+        elif [[ "${{ runner.os }}" == "macOS" ]]; then
+          # Install PowerShell on macOS
+          brew install --cask powershell
+        fi
+
     - name: Validate performance
       shell: pwsh
-      # Uses ETL which needs Windows
-      if: ${{ runner.os == 'Windows' }} 
+      # Now runs on all platforms, with ETL disabled on non-Windows
+      if: ${{ runner.os == 'Linux' }} 
```

Comment:
For now, make it run on both Linux and Windows. We will remove Windows when the OS is dropped from the matrix